### PR TITLE
backend: add "yaml" as a supported backend

### PR DESCRIPTION
**Pull request by davvid**
YAML is a superset of JSON and has many nice features such as
support for comments and a generally nicer syntax.

Add support for using "yaml" as a jsonpickle backend.

### CODE REVIEW

**Comment by davvid** [comment]
This uses changes from #527 so I'll rebase this after that's merged to make the diff in this PR easier to review.

File: jsonpickle/backend.py
Comment (Theelx) [inline jsonpickle/backend.py:19]: Thanks for updating this!

File: jsonpickle/backend.py
Comment (Theelx) [inline jsonpickle/backend.py:97]: If I recall correctly, aren't backends attempted in the order in which they are loaded? Might make sense to load ujson first, at least until people start using the yaml backend.

**Review by Theelx — APPROVED** [review]
Minor nitpick about yaml/ujson backend priority, but otherwise LGTM after the rebase!

File: jsonpickle/backend.py
Comment (davvid) [reply jsonpickle/backend.py:97]: True, sorry for not catching this until post-merge. I'll fix this up locally and push it out shortly.

**Pull request merged**
